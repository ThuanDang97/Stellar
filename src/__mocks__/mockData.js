export const TOOLTIP_LOREM =
  'Lorem Ipsum has been the industrys standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.'

export const SELECT_OPTIONS = [
  { value: '1', name: 'Pay monthly' },
  { value: '3', name: 'Pay 3 monthly' },
  { value: '6', name: 'Pay 6 monthly' },
  { value: '12', name: 'Pay annually' },
]

export const USER_FULL_NAME = 'John Wick'

export const AVATAR_DEFAULT = {
  firstName: 'John',
  lastName: 'Wick',
}

export const AVATAR_IMAGE = {
  imageUrl: 'https://picsum.photos/20',
}

export const NETFLIX_REPORT = {
  name: 'netflix',
  period: 'report monthly',
}

export const GOAL_ITEM = {
  title: 'Improve My Credit Score',
  iconUrl: '/icons/piggy-bank',
}

export const BANK_NUMBER = '123456789111'

export const BANK_NAME = 'Chase'

export const INPUT_VALUES = {
  email: 'test@gmail.com',
  text: 'example',
  password: 'example1password',
  image:
    'data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2716%27%20height=%2716%27/%3e',
}

export const MEDIA_IMAGES = {
  netflix: '/images/netflix.png',
  hulu: '/images/hulu.png',
}

export const ADD_BILL_LIST = [
  {
    id: '1',
    description: 'Netflix',
    amount: 50,
    buttonControl: '',
    imageUrl: MEDIA_IMAGES.netflix,
    created: 121212121,
  },
  {
    id: '2',
    description: 'Holo',
    amount: 77,
    buttonControl: '',
    imageUrl: MEDIA_IMAGES.hulu,
    created: 121212121,
  },
]

export const MY_GOALS = [
  {
    id: '1',
    goal: 'Buy a car',
    scoreNeeded: 703,
    pointNeeded: 'READY',
    imageUrl: MEDIA_IMAGES.netflix,
    created: 121212121,
  },
  {
    id: '2',
    goal: 'Buy a house',
    scoreNeeded: 720,
    pointNeeded: '23 PTS AWAY',
    imageUrl: MEDIA_IMAGES.hulu,
    created: 3434543645,
  },
]

export const LINKED_BILL_LIST = [
  {
    id: '1',
    description: 'Netflix',
    amount: 50,
    draftDate: '1st',
    status: 'Unpaid',
    imageUrl: MEDIA_IMAGES.netflix,
    created: 121212121,
  },
  {
    id: '2',
    description: 'Hulu',
    amount: 32,
    draftDate: '4th',
    status: 'Paid',
    imageUrl: MEDIA_IMAGES.hulu,
    created: 121212121,
  },
]

export const EXITS_EMAIL = 'thanh@endurance.com'

export const NUMBER_BANK = [
  {
    id: '1',
    bankNumber: '12435324234',
    bankName: 'Chase',
  },
]

export const ACCOUNT_LOGIN = [
  {
    userId: '1',
    email: 'khanhtam304n@gmail.com',
    password: 'Khanhtam@123',
  },
  {
    userId: '2',
    email: 'khanhtam3041n@gmail.com',
    password: 'Khanhtam@123',
  },
]

export const USER = {
  userId: '4',
  userName: 'Khanh Tam',
  email: 'khanhtam304n@gmail.com',
  password: 'Khanhtam@123',
}

export const GOAL = ['g1', 'g2']

export const MARKETING = ['m1', 'm2']

export const USER_ID = '123'

export const DATE_TIMER = new Date('08/28/2022')

export const USER_GOAL = { userId: USER_ID, goals: GOAL }

export const USER_MARKETING = { userId: USER_ID, goals: MARKETING }

export const ADDRESS = {
  address: '123 street',
  phone: '1231231231',
}

export const SSN_NUMBER = '123123123'

export const SUBSCRIPTION_PLAN = 'flex'

export const USER_SUBSCRIPTION_PLAN = {
  userId: USER_ID,
  subscriptionPlan: SUBSCRIPTION_PLAN,
}

export const CHART_SCORE = {
  data: [
    { month: 'M', score: 720 },
    { month: 'A', score: 740 },
    { month: 'M', score: 768 },
    { month: 'J', score: 785 },
    { month: 'J', score: 810 },
    { month: 'A', score: 830 },
  ],
}

export const LINK_TOKEN = 'sand-box-123'

export const BANK_ID = 'id_45'

export const BANK_INFO = {
  id: BANK_ID,
  bankNumber: BANK_NUMBER,
  bankName: BANK_NAME,
}
export const FICO_DATE_UPDATE = 'Last Updated Aug, 12, 2021'

export const NOTIFICATIONS = [
  {
    id: '1',
    title: 'Credit Score Update',
    desc: 'Your credit score increased by...',
    dateTime: '1 hours',
    isCheck: false,
  },
  {
    id: '2',
    title: 'Bill Pay',
    desc: 'We noticed you missed a payment...',
    dateTime: '3 days ago',
    isCheck: false,
  },
  {
    id: '3',
    title: 'Offer',
    desc: 'Look like you may quality for...',
    dateTime: '10 days ago',
    isCheck: false,
  },
  {
    id: '4',
    title: 'Offer',
    desc: 'Look like you may quality for offer',
    dateTime: '11 days ago',
    isCheck: false,
  },
]

export const APPLICATION_INFO = {
  title: 'Netflix',
  image: '/images/netflix.png',
}

export const APPLICATION_CONNECT_ACCOUNT = {
  userId: '1',
  email: 'khanhtam304n@gmail.com',
  password: 'Khanhtam@123',
}

// Min score: 300 - Max score 850
export const FICO_SCORE = 500

export const IDENTITY = {
  firstName: 'John',
  lastName: 'Wick',
  dateOfBirth: '02/02/2000',
  ssn: 123123123,
}

export const SSN_USER = {
  userId: USER_ID,
  ssn: SSN_NUMBER,
}

export const USER_IDENTITY = {
  userId: USER_ID,
  identity: IDENTITY,
}

export const ADDRESS_USER = {
  address: '604 Nui Thanh',
  phoneNumber: '+10949328492',
}

export const FICO_UNLOCK_SCORE = 708

export const LIST_BANKS_NUMBER = [
  { id: 'Chase', bankNumber: '423432432423232' },
  { id: 'Bank of America', bankNumber: '111122223333000' },
  { id: 'CitiBank Online', bankNumber: '111122223333111' },
]

export const INFO_PAYMENTS = {
  id: '123',
  money: '15.99',
  bankName: 'Chase',
  payDate: new Date(),
  numberPaymentsMonth: '3',
  appInfo: {
    title: 'Netflix',
    image: '/images/netflix.png',
  },
  status: 'UNPAID',
}

export const BILL_INFO = {
  title: 'Netflix',
  amount: 10,
}

export const ADDED_BILL_ID = '1234'

export const USER_DETAIL = {
  id: '123',
  amount: 10,
  draftDate: 31,
  bankNumber: BANK_NUMBER,
}

export const EDIT_LINK_BILL = {
  id: '123',
  amount: 10,
}
