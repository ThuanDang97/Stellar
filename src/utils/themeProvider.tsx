import { ThemeProvider } from 'styled-components'
import { render } from '@testing-library/react'
import { theme } from '@themes/index'

export const renderWithTheme = (component: React.ReactNode) => {
  return render(<ThemeProvider theme={theme}>{component}</ThemeProvider>)
}
