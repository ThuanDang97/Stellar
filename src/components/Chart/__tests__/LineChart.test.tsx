// Components
import LineChart from '@components/Chart'

// Utils
import { renderWithTheme } from '@utils/themeProvider'

// Props type
import { LineChartProps } from '@self-types/components/LineChart.props'
import { CHART_SCORE } from '@mocks/mockData'

jest.mock('react-chartjs-2', () => ({
  Line: () => null,
}))
jest.mock('chartjs-plugin-datalabels', () => ({
  ChartDataLabels: [],
}))
jest.mock('chart.js', () => ({
  Chart: { register: jest.fn(), defaults: { set: jest.fn() } },
  registerables: [],
}))

describe('Line Chart Component', () => {
  const props: LineChartProps = {
    isLocked: false,
    dataChart: CHART_SCORE.data,
  }

  const secondaryProps: LineChartProps = {
    isLocked: true,
    dataChart: [],
  }

  const customProps: LineChartProps = {
    isLocked: false,
    dataChart: [],
  }

  test('renders Line Chart have data unchanged', () => {
    const { container } = renderWithTheme(<LineChart {...props} />)

    expect(container).toMatchSnapshot()
  })

  test('renders Line Chart is locked unchanged', () => {
    const { container } = renderWithTheme(<LineChart {...secondaryProps} />)

    expect(container).toMatchSnapshot()
  })

  test('renders Line Chart unlock and no data unchanged', () => {
    const { container } = renderWithTheme(<LineChart {...customProps} />)

    expect(container).toMatchSnapshot()
  })
})
