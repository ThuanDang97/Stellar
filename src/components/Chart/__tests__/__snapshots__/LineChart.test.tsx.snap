// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Line Chart Component renders Line Chart have data unchanged 1`] = `
<div>
  <div
    class="ViewStyled-sc-rqe5sa-0 ViewStyled__BoxShadowStyled-sc-rqe5sa-2 hcqEVE dPpNOf"
    display="flex"
    width="100%"
  >
    <h1
      class="TextStyled-sc-9xia41-0 TitleStyled-sc-t58pjn-0 eJbYjT ivXuze"
      color="#000"
      font-size="18"
    >
      Score Projection
    </h1>
    <div
      class="ViewStyled-sc-rqe5sa-0 ksGJjB"
      height="250"
      width="100%"
    />
  </div>
  <div
    class="ViewStyled-sc-rqe5sa-0 eWEXdJ"
    display="flex"
  >
    <a
      class="LinkStyled__ButtonLinkStyled-sc-626068-1 cYfvgi"
      color="#919099"
      data-testid="link"
      font-size="13"
      href="/privacy-policy"
      letter-spacing="0.9"
    >
      Privacy Policy
    </a>
    <span
      class="TextStyled-sc-9xia41-0 hrVQLR"
      color="#919099"
    >
       - 
    </span>
    <a
      class="LinkStyled__ButtonLinkStyled-sc-626068-1 cYfvgi"
      color="#919099"
      data-testid="link"
      font-size="13"
      href="/terms-of-use"
      letter-spacing="0.9"
    >
      Terms of Use
    </a>
  </div>
</div>
`;

exports[`Line Chart Component renders Line Chart is locked unchanged 1`] = `
<div>
  <div
    class="ViewStyled-sc-rqe5sa-0 ViewStyled__BoxShadowStyled-sc-rqe5sa-2 hcqEVE dPpNOf"
    display="flex"
    width="100%"
  >
    <h1
      class="TextStyled-sc-9xia41-0 TitleStyled-sc-t58pjn-0 eJbYjT ivXuze"
      color="#000"
      font-size="18"
    >
      Score Projection
    </h1>
    <div
      class="ViewStyled-sc-rqe5sa-0 ksGJjB"
      height="250"
      width="100%"
    />
  </div>
  <div
    class="ViewStyled-sc-rqe5sa-0 eWEXdJ"
    display="flex"
  >
    <a
      class="LinkStyled__ButtonLinkStyled-sc-626068-1 cYfvgi"
      color="#919099"
      data-testid="link"
      font-size="13"
      href="/privacy-policy"
      letter-spacing="0.9"
    >
      Privacy Policy
    </a>
    <span
      class="TextStyled-sc-9xia41-0 hrVQLR"
      color="#919099"
    >
       - 
    </span>
    <a
      class="LinkStyled__ButtonLinkStyled-sc-626068-1 cYfvgi"
      color="#919099"
      data-testid="link"
      font-size="13"
      href="/terms-of-use"
      letter-spacing="0.9"
    >
      Terms of Use
    </a>
  </div>
</div>
`;

exports[`Line Chart Component renders Line Chart unlock and no data unchanged 1`] = `
<div>
  <div
    class="ViewStyled-sc-rqe5sa-0 ViewStyled__BoxShadowStyled-sc-rqe5sa-2 hcqEVE dPpNOf"
    display="flex"
    width="100%"
  >
    <h1
      class="TextStyled-sc-9xia41-0 TitleStyled-sc-t58pjn-0 eJbYjT ivXuze"
      color="#000"
      font-size="18"
    >
      Score Projection
    </h1>
    <div
      class="ViewStyled-sc-rqe5sa-0 ksGJjB"
      height="250"
      width="100%"
    >
      <div
        class="ViewStyled-sc-rqe5sa-0 jRwnka"
        width="120"
      >
        <span
          class="TextStyled-sc-9xia41-0 iWmMQb"
          color="#000"
          font-size="13"
          letter-spacing="0.68"
        >
          Add Bills To Project Your Future Score
        </span>
      </div>
    </div>
  </div>
  <div
    class="ViewStyled-sc-rqe5sa-0 eWEXdJ"
    display="flex"
  >
    <a
      class="LinkStyled__ButtonLinkStyled-sc-626068-1 cYfvgi"
      color="#919099"
      data-testid="link"
      font-size="13"
      href="/privacy-policy"
      letter-spacing="0.9"
    >
      Privacy Policy
    </a>
    <span
      class="TextStyled-sc-9xia41-0 hrVQLR"
      color="#919099"
    >
       - 
    </span>
    <a
      class="LinkStyled__ButtonLinkStyled-sc-626068-1 cYfvgi"
      color="#919099"
      data-testid="link"
      font-size="13"
      href="/terms-of-use"
      letter-spacing="0.9"
    >
      Terms of Use
    </a>
  </div>
</div>
`;
